zebra-network/src/config.rs

```diff
@@ -9,10 +16,11 @@ use crate::BoxError;
 const MAX_SINGLE_PEER_RETRIES: usize = 2;
 
 /// Configuration for networking code.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-#[serde(deny_unknown_fields, default)]
+#[derive(Clone, Debug, Serialize)]
 pub struct Config {
     /// The address on which this node should listen for connections.
+    /// Can be `ip:port` or just `ip` (in this case `port` will be derived
+    /// from `network`).
```

Comment:
We also accept DNS names, tweak config doc style, add missing info:
```suggestion
    ///
    /// Can be `address:port` or just `address`. If there is no configured
    /// port, Zebra will use the default port for the configured `network`.
    /// `address` can be an IP address or a DNS name. DNS names are
    /// only resolved once, when Zebra starts up.
```